app/concerns/soft_deletion.rb

```diff
@@ -0,0 +1,25 @@
+require 'active_support/concern'
+
+module SoftDeletion
+  extend ActiveSupport::Concern
+
+  included do
+    default_scope { where(deleted_at: nil) }
+
+    scope :only_deleted, -> { unscope(where: :deleted_at).where.not(deleted_at: nil) }
+    scope :with_deleted, -> { unscope(where: :deleted_at) }
+
+    def soft_delete
+      update(deleted_at: Time.now)
+    end
+
+    def soft_deleted?
+      deleted_at != nil
+    end
+
+    # would like to be able to skip any validation
+    def restore
+      update(deleted_at: nil)
+    end
+  end
+end
```

Comment:
I would put model concerns in `app/model/concerns`.

Reply:
I created an issue to move the remaining files out of app/concerns https://github.com/sudara/alonetone/tree/master/app/concerns